kong/plugins/ip-restriction/handler.lua

```diff
@@ -1,22 +1,44 @@
+local lrucache = require "resty.lrucache"
 local ipmatcher = require "resty.ipmatcher"
 local kong_meta = require "kong.meta"
 
 
-local ngx = ngx
+local ngx_var = ngx.var
 local kong = kong
 local error = error
 
 
+local IPMATCHER_COUNT = 512
+local cache = lrucache.new(IPMATCHER_COUNT)
+
+
 local IpRestrictionHandler = {
   PRIORITY = 990,
   VERSION = kong_meta.version,
 }
 
 
+local isempty
+do
+  local tb_isempty = require "table.isempty"
+
+  isempty = function(t)
+    return t == nil or tb_isempty(t)
+  end
+end
+
+
 local function match_bin(list, binary_remote_addr)
-  local ip, err = ipmatcher.new(list)
-  if err then
-    return error("failed to create a new ipmatcher instance: " .. err)
+  local ip, err
+
+  ip = cache:get(list)
```

Comment:
This has no purging when the config changes. Will this cause cache poisoning?

Reply:
I also considered this problem, but in our test case `03-plugins/17-ip-restriction/02-access_spec.lua`,
we test it with `PATCH` method, does it make sense?

Reply:
Or we add a ttl to cache, like 60 seconds?